dashboard/app/models/user.rb

```diff
@@ -490,10 +490,34 @@ def self.authenticate_with_section_and_secret_picture(section:, params:)
       )
   end
 
+  # cache and return (from cache, if present) all user levels for this script for this user.
+  # The cache lifetime is the current HTTP request.
+  def cached_user_levels_by_script(script_id)
+    @user_level_cache_by_script ||= {}
+    if @user_level_cache_by_script[script_id].nil?
+      # make a database dip to get user_levels
+      @user_level_cache_by_script[script_id] =
+        user_levels.where(script_id: script_id).index_by(&:level_id)
+    end
+    @user_level_cache_by_script[script_id]
+  end
+
+  # returns if all user levels for this script have been cached
+  def user_levels_cached_for_script?(script_id)
+    @user_level_cache_by_script.try(:[], script_id).nil? ? false : true
```

Comment:
This could be simplified to `!@user_level_cache_by_script.try(:[], script_id).nil?`?